idaes/models/control/tests/test_antiwindup.py

```diff
@@ -0,0 +1,571 @@
+from idaes.models.properties import iapws95
+from idaes.core.util.initialization import propagate_state
+from idaes.models.control.controller import (
+    PIDController,
+    ControllerType,
+    ControllerMVBoundType,
+    ControllerAntiwindupType,
+)
+import idaes.core.util.scaling as iscale
+from idaes.core.util.math import safe_sqrt
+from idaes.core.solvers import get_solver, petsc
+import idaes.logger as idaeslog
+from idaes.core.util.plot import plot_grid_dynamic
+import idaes.core.util.scaling as iscale
+
+
+def set_indexed_variable_bounds(var, bounds):
+    for idx, subvar in var.items():
+        subvar.bounds = bounds
+
+
+def _valve_pressure_flow_cb(b):
+    """
+    For vapor F = Cv*sqrt(Pi**2 - Po**2)*f(x)
+    """
+    umeta = b.config.property_package.get_metadata().get_derived_units
+
+    b.Cv = pyo.Var(
+        initialize=0.1,
+        doc="Valve flow coefficent",
+        units=umeta("amount") / umeta("time") / umeta("pressure"),
+    )
+    b.Cv.fix()
+
+    b.flow_var = pyo.Reference(b.control_volume.properties_in[:].flow_mol)
+    # b.pressure_flow_equation_scale = lambda x: x ** 2
+    b.pressure_flow_equation_scale = lambda x: x
+
+    @b.Constraint(b.flowsheet().time)
+    def pressure_flow_equation(b2, t):
+        Po = b2.control_volume.properties_out[t].pressure
+        Pi = b2.control_volume.properties_in[t].pressure
+        F = b2.control_volume.properties_in[t].flow_mol
+        Cv = b2.Cv
+        fun = b2.valve_function[t]
+        # return F**2 == Cv**2 * (Pi**2 - Po**2) * fun**2
+        # return F == Cv * pyo.sqrt(Pi ** 2 - Po ** 2) * fun
+        # return F ** 2 == Cv ** 2 * (Pi + Po) * (-b2.deltaP[t]) * fun ** 2
+        return F == Cv * pyo.sqrt((Pi + Po) * (-b2.deltaP[t])) * fun
+
+
+def _add_inlet_pressure_step(m, time=1, value=6.0e5):
+    """Easy function to add an inlet pressure step change
+
+    Args:
+        m (ConcreteModel): A valve and tank model
+        time (float): Time for the step to occur
+        value (float): New pressure at time
+
+    Returns:
+        None
+    """
+    for t in m.fs.time:
+        if t >= time:
+            m.fs.valve_1.inlet.pressure[t].fix(value)
+
+
+def _add_inlet_pressure_step(m, time=1, value=6.0e5):
+    """Easy function to add an inlet pressure step change
+
+    Args:
+        m (ConcreteModel): A valve and tank model
+        time (float): Time for the step to occur
+        value (float): New pressure at time
+
+    Returns:
+        None
+    """
+    for t in m.fs.time:
+        if t >= time:
+            m.fs.valve_1.inlet.pressure[t].fix(value)
+
+
+def _add_setpoint_step(m, time=1, value=6.0e5):
+    """Easy function to add an inlet pressure step change
+
+    Args:
+        m (ConcreteModel): A valve and tank model
+        time (float): Time for the step to occur
+        value (float): New pressure at time
+
+    Returns:
+        None
+    """
+    for t in m.fs.time:
+        if t >= time:
+            m.fs.ctrl.setpoint[t].fix(value)
```

Comment:
This function is also practically a duplicate of the one above. Why not just add an additional argument that accepts the time-indexed var to fix in the for-loop and then collapse these to a single function?